content/blog/open-sourcing-workshops-on-demand-part-2-how-to-deploy-the-infrastructure.md

```diff
@@ -0,0 +1,499 @@
+
+I﻿ will start with the simpliest scenario: A public-only approach. Then we will dive into the specificities related the private approach.
+
+### P﻿ublic-only Deployment: No private backend nor private workshops
+
+**Important Note:**  
+
+**T﻿his part is compulsory for any type of deployment. Public only or public + private.**
+
+F﻿irst, you need a repository to clone. The Workshops-on-Demand GitHub projects can be found [here](https://github.com/Workshops-on-Demand/). W﻿e have packaged the solution in several Github repos. Each repository handles a specific role in the overall architecture.
+
+Here's a quick look at what can be found in each:
+
+![](/img/wod-repository-2025.png "WOD Repositories")
+
+**[w﻿od-notebooks](https://github.com/Workshops-on-Demand/wod-notebooks):** Public Workshops-on-Demand based on Jupyter Notebooks.
+
+* You can test them live at <https://hackshack.hpedev.io/workshops>
+
+**[w﻿od-install](https://github.com/Workshops-on-Demand/wod-install):** Installer part of the Workshops-on-Demand project.
+
+**[w﻿od-backend](https://github.com/Workshops-on-Demand/wod-backend):** Back-end part of our Workshops-on-Demand setup. 
+
+**[w﻿od-frontend](https://github.com/Workshops-on-Demand/wod-frontend):** Frontend part of the Workshops-on-Demand project.
+
+* Based on NGINX and NodeJS technologies, it provides the participtants' Registration Portal used to enable booking of the workshops.
+
+**[w﻿od-api-db](https://github.com/Workshops-on-Demand/wod-api-db):** Workshops-on-Demand registration portal application
+
+* Open API 3.0 based api used to manage the Workshops-on-Demand project. It also provides a database hosting the different status of participants, workshops, and students. 
+
+**[w﻿od-private](https://github.com/Workshops-on-Demand/wod-private):** Example Private configuration for Workshops-on-Demand (WoD).
+
+**[w﻿od-frontend-private](https://github.com/Workshops-on-Demand/wod-frontend-private):** Private Frontend part of the Workshops-on-Demand project.
+
+**[w﻿od-api-db-private](https://github.com/Workshops-on-Demand/wod-api-db-private):** Workshops-on-Demand registration portal application
+
+* This provide examples for creating your own cutomization layer on top of the public standard WoD Backend / wod Notebooks content. Do not put any confidential data here as this is a public repository!
+
+**Note**: T﻿here are now 9 repositories available for now. 
+
+![](/img/wod-blogserie2-2repos.png "Workshops-on-Demand repositories")
+
+It provides:
+
+* An Installer allowing you to install either Backend, Api-DB server, or Frontend using a single line of command.
+* A complete JupyterHub server with some addons (additional Jupyterhub kernels, Ansible galaxies, and PowerShell libraries) on your system, ready to host Workshops-on-Demand that you can find here. 
+* A postfix server used for the procmail API 
+* An Ansible engine to allow automation 
+* A fail2ban service 
+* An Admin user to manage everything 
```

Comment:
You find an Admin user in there? How do you package the person in there? Must be a very tiny person...